### PR TITLE
[#1041] Split cascade data based on `useQuestionId` flag

### PR DESCRIPTION
**Pull request by iperdomo**

### CODE REVIEW

**Comment by iperdomo** [comment]
## Checklist

* [ ] Test plan
* [ ] Copyright header
* [ ] Code formatting

File: GAE/src/org/waterforpeople/mapping/dataexport/GraphicalSurveySummaryExporter.java [hunk 1]
```diff
@@ -701,10 +705,17 @@ private synchronized void writeRow(Row row, SurveyInstanceDto dto,
...
+                            if (useQuestionId) {
+                                createCell(row, offset++,
+                                        (useQID ? questionId + "_" : q.getText() + " - ")
+                                                + LAT_LABEL.get(columnLocale),
+                                        headerStyle);
+                            } else {
+                                createCell(row, offset++,
+                                        (useQID ? questionId + "_" : q.getKeyId() + "|")
+                                                + LAT_LABEL.get(columnLocale),
+                                        headerStyle);
```
Comment (muloem) [inline GAE/src/org/waterforpeople/mapping/dataexport/GraphicalSurveySummaryExporter.java]: In this second branch of the if statement the first option of the ternary `useQID ? questionId + "_" :` will never be executed no? Maybe you can just output the `keyId` with the pipe.

File: GAE/src/org/waterforpeople/mapping/dataexport/GraphicalSurveySummaryExporter.java [hunk 2]
```diff
@@ -718,14 +729,23 @@ private synchronized void writeRow(Row row, SurveyInstanceDto dto,
                                     useQID ? questionId + "_" + codeLabel.replaceAll("\\s", "")
                                             : "--GEOCODE--|" + codeLabel,
                                     headerStyle);
-                        } else if (QuestionType.CASCADE == q.getType() && q.getLevelNames() != null) {
+                        } else if (QuestionType.CASCADE == q.getType() && q.getLevelNames() != null
+                                && useQuestionId) {
                             for (String level : q.getLevelNames()) {
-                                createCell(row, offset++, level, headerStyle);
+                                String levelName = useQID ? questionId + "_"
+                                        + level.replaceAll(" ", "_") : q.getText()
```
Comment (muloem) [inline GAE/src/org/waterforpeople/mapping/dataexport/GraphicalSurveySummaryExporter.java]: I see you have used `getLocalizedText()` in other parts. Maybe here as well?

File: GAE/src/org/waterforpeople/mapping/dataexport/GraphicalSurveySummaryExporter.java [hunk 3]
```diff
@@ -701,10 +705,17 @@ private synchronized void writeRow(Row row, SurveyInstanceDto dto,
                         String columnLocale = useQID ? "en" : locale;
 
                         if (QuestionType.GEO == q.getType()) {
-                            createCell(row, offset++,
-                                    (useQID ? questionId + "_" : q.getKeyId().toString() + "|")
-                                            + LAT_LABEL.get(columnLocale),
-                                    headerStyle);
+                            if (useQuestionId) {
+                                createCell(row, offset++,
+                                        (useQID ? questionId + "_" : q.getText() + " - ")
```
Comment (muloem) [inline GAE/src/org/waterforpeople/mapping/dataexport/GraphicalSurveySummaryExporter.java]: same comment regarding localized text?

**Comment by muloem** [comment]
:+1: code reviewed!

**Pull request merged**